client/components/BudgetGrid/EntryFormRow.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component, PropTypes } from 'react';
+
+class EntryFormRow extends Component {
+  static propTypes = {
+    categories: PropTypes.object.isRequired
+  }
+
+  state = {
+    categoryId: 16,
+    description: '',
+    amount: ''
+  }
+
+  getCategoryOptions() {
```

Comment:
Yep, I thought I had smelled this earlier but didn't take action. I have moved EntryFormRow to its own pod, with a new component named DataSelector underneath.